# named list of all robots
robots:
  cf1:
    enabled: true
    uri: radio://0/80/2M/E7E7E7E701
    # uri: radio://0/80/2M/E7E7E7E702
    initial_position: [0.0, 0.0, 0.0]
    type: cf21  # see robot_types
  cf2:
    enabled: false
    uri: radio://0/80/2M/E7E7E7E702
    # uri: radio://0/80/2M/E7E7E7E702
    initial_position: [0.0, 0.0, 0.0]
    type: cf21  # see robot_types
# Definition of the various robot types
robot_types:
  cf21:
    motion_capture:
      enabled: true
      # only if enabled; see motion_capture.yaml
      marker: default_single_marker
      dynamics: default
    big_quad: false
    battery:
      voltage_warning: 4.0  # V
      voltage_critical: 3.9 # V
    firmware_params:
      ctrlMel:
        kR_xy: 1 #300
        kR_z: 0 #150

# global settings for all robots
all:
  # firmware logging for all drones (use robot_types/type_name to set per type, or
  # robots/drone_name to set per drone)
  firmware_logging:
    enabled: true
    default_topics:
     # remove to disable default topic
      pose:
        frequency: 50 # Hz
      odom:
        frequency: 50 # Hz
    #custom_topics:
    #  topic_name1:
    #    frequency: 10 # Hz
    #    vars: ["stateEstimateZ.x", "stateEstimateZ.y", "stateEstimateZ.z", "pm.vbat"]
    #  topic_name2:
    #    frequency: 1 # Hz
    #    vars: ["stabilizer.roll", "stabilizer.pitch", "stabilizer.yaw"]
  # firmware parameters for all drones (use robot_types/type_name to set per type, or
  # robots/drone_name to set per drone)
  firmware_params:
    commander:
      enHighLevel: 1
    stabilizer:
      estimator: 2 # 1: complementary, 2: kalman
      controller: 1 # 1: PID, 2: mellinger 3:INDI 4: brescianini
    locSrv:
      extPosStdDev: 1e-3
      extQuatStdDev: 0.5e-1
    kalman:
      resetEstimation: 1
    ctrlAtt:
      kp_rp_rate: 150.0   # 150.0 works good for hover
      ki_rp_rate: 20.0    # 20.0 works good for hover
      kp_yaw_rate: 70.0
      ki_yaw_rate: 15.0
      omega_yaw_i_max: 30.0
      omega_rp_i_max: 500.0
  broadcasts:
    num_repeats: 15 # number of times broadcast commands are repeated
    delay_between_repeats_ms: 1 # delay in milliseconds between individual repeats
